arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/stream/concurrent/QueueTest.kt

```diff
@@ -188,4 +192,94 @@ class QueueTest : StreamSpec(spec = {
       q.dequeue1() shouldBe 1
     }
   }
+
+  "Queue.bounded with capacity can always take tryOffer1" {
+    checkAll(Arb.int()) { i ->
+      val q = Queue.bounded<Int>(1)
+
+      q.tryOffer1(i) shouldBe true
+      q.dequeue().take(1).toList() shouldBe listOf(i)
+    }
+  }
+
+  "Queue.bounded with outstanding taker received tryOffer1 value" {
+    checkAll(Arb.int()) { i ->
+      val q = Queue.bounded<Int>(1)
+      val start = Promise<Unit>()
+
+      val f = ForkAndForget {
+        start.complete(Unit)
+        q.dequeue1()
+      }
+
+      start.get()
+
+      q.tryOffer1(i) shouldBe true
+      f.join() shouldBe i
+    }
+  }
+
+  "Queue.bounded producer/consumer" {
+    checkAll(Arb.stream(Arb.int())) { s ->
+      val expected = s.toList()
+      val n = expected.size
+      val q = Queue.bounded<Int>(n)
+
+      Stream(
+        q.dequeue(),
+        s.through(q.enqueue()).void()
+      ).parJoinUnbounded()
+        .take(n)
+        .toList() shouldBe expected
+    }
+  }
+
+  "Queue.bounded none terminating producer/consumer" {
+    checkAll(Arb.stream(Arb.int())) { s ->
+      val expected = s.toList()
+
+      // Queue.bounded requires minimum size of 1
+      val n = max(expected.size, 1)
+      val q = Queue.bounded<Option<Int>>(n)
+
+      q.dequeue()
+        .terminateOnNone()
+        .concurrently(
+          s.noneTerminate()
+            .through(q.enqueue())
+        ).toList() shouldBe s.toList()
+    }
+  }
+
+  "Queue.bounded(0) with outstanding taker cannot offer" {
+    val q = Queue.bounded<Int>(0)
+    ForkConnected { q.dequeue1() }
```

Comment:
Is this needed?

Reply:
That is to test that even tho someone is trying to take from the `Queue`, that `offer` still fails. We can't assume from the test that it won't do that, right? 😅